strapi-app/src/api/line/services/line.js

```diff
@@ -50,6 +49,35 @@ module.exports = {
     }
   },
 
+  async messagingEvent(event) {
+    if (!event?.source?.userId) return;
+    try {
+      const context = await strapi
+                  .query("api::line-user.line-user")
+                  .findOne({select: ['context'], where: { user_id: event.source.userId }});
+      if(context.context === 'name_setting'){
+        if (event.message.text.length >= 11) {
+          var display_name = event.message.text.substr( 0, 10 );
+        }
+        else{
+          var display_name = event.message.text;
+        }
```

Comment:
動作に問題はないけどもう少し簡単に書いてみましょうか〜
```suggestion
        const text = event.message.text;
        const display_name = text.length < 11 ? text : text.substr(0, 10);
```